File: Ejercicios_etapa_1/Lua/Ejercicio_8/modules/get_product_handles_and_ids.lua
-- modules/get_product_handles_and_ids.lua
-- Función que combina IDs y Handles en una nueva tabla

-- TODO: importa las funciones desde utils.lua
-- local utils = require("utils")

local function getProductHandlesAndIds(products)
    local handles = utils.getProductHandles(products)
    local ids = utils.getProductIds(products)

    local resultado = {}
    for i = 1, #products do
        table.insert(resultado, {
            id = ids[i],
            handle = handles[i]
        })
    end

    return resultado
end

-- Aquí debe ir el return de getProductHandlesAndIds

File: Ejercicios_etapa_1/00_Rust/importar/src/get_product_handles_and_ids.rs
// src/get_product_handles_and_ids.rs

// TODO: Importa `get_product_handles` y `get_product_ids` desde utils.rs

use crate::Product; // Importamos la estructura Product

pub fn get_product_handles_and_ids(products: &[Product]) -> Vec<(u32, String)> {
    let handles = get_product_handles(products);
    let ids = get_product_ids(products);

    ids.into_iter()
        .zip(handles.into_iter())
        .collect()
}
